iothub/device/src/PropertyCollection.cs

```diff
@@ -112,10 +116,10 @@ private void Add(IDictionary<string, object> properties, string componentName =
                 }
                 foreach (KeyValuePair<string, object> entry in properties)
                 {
-                    var checkType = entry.Value is WritablePropertyBase;
-                    if (entry.Value is WritablePropertyBase && !Convention.PayloadSerializer.CheckWritablePropertyResponseType(entry.Value))
+                    bool checkType = entry.Value is IWritablePropertyResponse;
```

Comment:
and this one

Reply:
it is repeated in the statement below, true